client/services/FeedbackGeneratorService.js

```diff
@@ -373,21 +410,46 @@ tie.factory('FeedbackGeneratorService', [
        * @returns {Feedback}
        */
       getFeedback: function(tasks, codeEvalResult, rawCodeLineIndexes) {
+        _resetCounters('consecutiveSameErrorCounter');
+
+        if (previousErrorString === codeEvalResult.getErrorString()) {
+          consecutiveSameErrorCounter++;
+        }
+
         var feedback = _getFeedbackWithoutReinforcement(
           tasks, codeEvalResult, rawCodeLineIndexes);
         if (tasks.length > 0 && !codeEvalResult.getErrorString()) {
           feedback.setReinforcement(
             ReinforcementGeneratorService.getReinforcement(
               tasks[tasks.length - 1], codeEvalResult));
         }
+        if (consecutiveSameErrorCounter === UNFAMILIARITY_THRESHOLD) {
+          feedback.appendTextParagraph(_getUnfamiliarLanguageFeedback());
+          consecutiveSameErrorCounter = 0;
+        }
+        previousErrorString = codeEvalResult.getErrorString();
         return feedback;
       },
       getSyntaxErrorFeedback: function(errorString) {
```

Comment:
This isn't a big deal but we should probably add a JSDoc comment for this function and in it we can explain some of the reasoning for incrementing and resetting the variables.